tracing-attributes/src/lib.rs

```diff
@@ -146,6 +146,17 @@ use syn::{
 /// }
 /// ```
 ///
+/// If the function returns a `Result<T, E>`, you can add `err` to emit error events when the
+/// function returns `Err`:
+///
+/// ```
+/// # use tracing_attributes::instrument;
+/// #[instrument(err)]
+/// fn my_function(arg: usize) -> Result<(), std::io::Error> {
+///     Ok(())
+/// }
```

Comment:
I haven't thought through the implications of this yet, but I _think_ this should be the default behavior for `#[instrument]`. Are there circumstances where we _don't_ want an `#[instrument]`ed method/function _not_ emitting an `ERROR`-verbosity event on it returning an error?

(I'm trying to think through what I know about bottlerocket...)

Reply:
1. Not all functions return `Result`, and functions that return `Result` may not necessarily call the type `Result`.
2. If you instrument every function in your call stack and a return propagates through all of them, you will get duplicate traces. (This is what's currently happening in my project but I'm fine with it? I imagine it's not ideal though.)

Both of these are probably solvable outside the scope of this PR.

Reply:
>     2. If you instrument every function in your call stack and an `Err` propagates through all of them, you will get duplicate traces. (This is what's currently happening in my project but I'm fine with it? I imagine it's not ideal though.)

Yeah, I think that for this reason, we probably want this behavior to be opt-in rather than opt-out.